Libraries/Components/StatusBar/StatusBar.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule StatusBar
+ * @flow
+ */
+'use strict';
+
+const React = require('React');
+const ColorPropType = require('ColorPropType');
+const Platform = require('Platform');
+
+const processColor = require('processColor');
+
+if (Platform.OS === 'ios') {
+  var RCTStatusBarManager = require('NativeModules').StatusBarManager;
+} else if (Platform.OS === 'android') {
+  var RCTStatusBarManager = require('NativeModules').StatusBarAndroid;
+}
+
+import type ReactElement from 'ReactElement';
+
+type DefaultProps = {
+  animated: boolean;
+};
+
+const StatusBar = React.createClass({
+  statics: {
+    _propsStack: [],
+  },
+
+  propTypes: {
+    /**
+     * If the status bar is hidden.
+     */
+    hidden: React.PropTypes.bool,
+    /**
+     * If the transition between status bar property changes should be animated.
+     * Supported for color, barStyle and hidden.
+     */
+    animated: React.PropTypes.bool,
```

Comment:
While I mostly agree with your decision about keeping this as an "animated" prop, We'll need to add another prop to define the transition type for showing/hiding on iOS to avoid loss of functionality. Maybe:

    /*
     * The transition effect when showing and hiding the status bar. Defaults to 'fade'
     * @platform ios
     */
    showHideTransition: React.PropTypes.oneOf([
      'fade',
      'slide',
    ]),